gapic-generator/lib/gapic/model/method/http_annotation.rb

```diff
@@ -35,122 +43,181 @@ class HttpAnnotation
         end
 
         ##
-        # Whether a method path is present and non-empty
+        # The implicit routing parameter names and their corresponding paterns,
+        # including the `*` pattern implied for the named segments
+        # without an explicitly specified pattern.
         #
-        # @return [Boolean]
-        def path?
-          !path.empty?
+        # @return [Array<Array<String>>]
+        def routing_params_with_patterns
+          bindings.first&.routing_params_with_patterns || []
         end
 
         ##
-        # A method path or an empty string if not present
+        # A single Http binding.
         #
-        # @return [String]
-        def path
-          return "" unless verb_path
-          verb_path[1]
-        end
+        class HttpBinding
+          ##
+          # @param binding [::Google::Api::HttpRule]
+          #   The baseline instance of the Http annotation
+          def initialize binding
+            @binding = binding
+          end
 
-        ##
-        # Whether any routing params are present
-        #
-        # @return [Boolean]
-        def routing_params?
-          routing_params.any?
-        end
+          ##
+          # Whether a http verb is present for this method
+          #
+          # @return [Boolean]
+          def verb?
+            !verb.nil?
+          end
 
-        ##
-        # The segment key names and their corresponding paterns,
-        # including the `*` pattern implied for the named segments
-        # without pattern explicitly specified.
-        #
-        # @return [Array<Array<String>>]
-        def routing_params_with_patterns
-          @routing_params_with_patterns ||= begin
-            Gapic::UriTemplate.parse_arguments(path).map do |name, pattern|
-              [name, pattern.empty? ? "*" : pattern]
+          ##
+          # The http verb for this method
+          #
+          # @return [Symbol, nil]
+          def verb
+            verb_path[0] if verb_path
+          end
+
+          ##
+          # Whether a method path is present and non-empty
+          #
+          # @return [Boolean]
+          def path?
+            !path.empty?
+          end
+
+          ##
+          # A method path or an empty string if not present
+          #
+          # @return [String]
+          def path
+            return "" unless verb_path
+            verb_path[1]
+          end
+
+          ##
+          # Whether any routing params are present
+          #
+          # @return [Boolean]
+          def routing_params?
+            routing_params.any?
+          end
+
+          ##
+          # The segment key names and their corresponding paterns,
+          # including the `*` pattern implied for the named segments
+          # without pattern explicitly specified.
+          #
+          # @return [Array<Array<String>>]
+          def routing_params_with_patterns
+            @routing_params_with_patterns ||= begin
```

Comment:
This begin/end is unnecessary.

Reply:
it's not but I find it looks better with a multiline block in ||= construction since the inner part gets a consistent indentation.

Reply:
I think Rubocop is okay with:

```ruby
            @routing_params_with_patterns ||=
              Gapic::UriTemplate.parse_arguments(path).map do |name, pattern|
                [name, pattern.empty? ? "*" : pattern]
              end
```

but... looking at it again, maybe you're right, it's a bit more readable with the begin/end.